code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -61,17 +63,32 @@ var/global/list/parasites = list() //all currently existing/living guardians
 		if("tech")
 			for(var/type in (subtypesof(/datum/guardianname/tech) - namedatum.type))
 				possible_names += new type
+		if("carp")
+			for(var/type in (subtypesof(/datum/guardianname/carp) - namedatum.type))
+				possible_names += new type
 	namedatum = pick(possible_names)
-	updatetheme()
+	updatetheme(pickedtheme)
 
-/mob/living/simple_animal/hostile/guardian/proc/updatetheme() //update the guardian's theme to whatever its datum is; proc for adminfuckery
+/mob/living/simple_animal/hostile/guardian/proc/updatetheme(theme) //update the guardian's theme to whatever its datum is; proc for adminfuckery
 	name = "[namedatum.prefixname] [namedatum.suffixcolour]"
 	real_name = "[name]"
 	icon_living = "[namedatum.parasiteicon]"
 	icon_state = "[namedatum.parasiteicon]"
 	icon_dead = "[namedatum.parasiteicon]"
 	bubble_icon = "[namedatum.bubbleicon]"
 
+	if (namedatum.stainself)
+		color = namedatum.colour
+
+	//Special case holocarp, because #snowflake code
+	if(theme == "carp")
```

Comment:
So, this isn't just adding holocarp, this is now turning into a refactor of the guardian code.

Can't we just merge this, and then I can get to work tidying it up? Because then if I fail to tidy it up, we still have holocarp.